### ISSUE TITLE
Requiring relative to subtype

### ISSUE THREAD

**Issue by miffels** [webpack/enhanced-require#2]
Hi sokra,

just a question (not an issue at all) I have been thinking about quite some time:

Following your [loadTemplate() example on the webpack page](https://github.com/webpack/webpack#a-small-example-whats-possible) I played around a bit with subclassing Backbone.View in order to achieve some tighter coupling with jade templates.

Currently I'm working with something like this

```
view
|__ JadeView.js
|__sub1
|  |__SpecificView1.js
|  |__template
|     |__SpecificView1.jade
|__sub2
   |__SpecificView2.js
   |__template
      |__SpecificView2.jade
```

That is, I have a couple of view sub types organized in sub folders, and each of those sub folders would have its own template directory in order to keep them neatly separated.
Of course I would not want to require the template of each of these specific views repeatedly but only once in JadeView.js, much like

```
// Somewhere in JadeView.js
require('./template/' + this.name + '.jade');
```

which would somehow refer to `view/sub1/template/SpecificView1.jade` when I create a new `SpecificView1` and `view/sub2/template/SpecificView2.jade` when I create a `SpecificView2` accordingly, and _not_ `view/template/<whatever>.jade`. I could not come up with a solution so far since I think it is totally normal that the above `require` call would _always_ look in the context of the file where it was defined, but is there some (automatic) workaround for that? For now I specify each subview's sub directory path manually when I declare it, so that `SpecificView1` basically tells `JadeView` it is in the `sub1` directory. Now that I am planning to move JadeView to a separate package, this would be impracticable though. Can you think of a way this could be solved?

**Comment by sokra** [comment]
Currently I cannot imagine a solution which is pretty.

But I'll have an idea, that it'll need some minor changes in webpack.

in `SpecificViewX.js`

``` javascript
var JadeView = require("./JadeView?" + __dirname);
```

in `JadeView.js`

``` javascript
var template = require(__resourceQuery + "/template/" + this.name + ".jade");
```

The change would be to evaluate __dirname and __resourceQuery as compile time constant.

A problem would be that the code of `JadeView.js` would be duplicated for each directory with SpecificViews.

**Comment by miffels** [comment]
Interesting, I also had that idea of evaluating `__dirname` and the like at compile time, but thought that would be too much of a hassle for even mentioning. Plus, you're right, even then I'd have to have a `JadeView` (or at least a subtype of it with that specific template loading part) in each of my project view root directories since it's still relative to that directory. I guess it's best, in that case, to let each `SpecificView` load the template on its own, i.e. just duplicate that template loading part. When I am going to move the `JadeView` type out of my projects in a separate package, then relative path snippets won't do anymore, unfortunately, since you never know where the files would be. On the other hand, I could barely still call it `JadeView` when it doesn't take care of the template loading, couldn't I :)

Thanks for your suggestions, though! Maybe I'll think about that some more later and let you know, if I have some major enlightenment.

Edit: Having a second look at the `JadeView` stub, would it really need to be replicated? if `__resourceQuery` contained the absolute directory path, then it should be just fine.

**Comment by sokra** [comment]
I made the changes to webpack... webpack/webpack@2183f05db8dd121f34a86114d949d7b27da22510

Now here is my idea: https://github.com/sokra/resourceQuery-example

**Comment by miffels** [comment]
Horrific! I can't wait to give it a try. Thanks a lot!
